googlemock/include/gmock/gmock-spec-builders.h

```diff
@@ -239,6 +243,8 @@ class GTEST_API_ UntypedFunctionMockerBase {
 
   // All expectations for this function mocker.
   UntypedExpectations untyped_expectations_;
+
+  bool registry_destructed_;
```

Comment:
Maybe we should just keep a Registry* instead of a bool.
That's the registry we'd have to unregister with when we die, and that registry can reset this pointer to nullptr when IT dies.